### DISCUSSION TITLE
Project request: micronaut-etcd

### DISCUSSION THREAD

**Discussion by marcosflobo** [micronaut-projects/micronaut-core#4486] [Ideas] [answered]
Dear community,

I'm opening this discussion to propose a new project within the micronaut umbrella.

### micronaut-etcd
The idea would be to create a new project to embed the Java client to manage the etcd API from micronaut applications, as we have to today for other important projects such as Redis, Consul, PosgreSQL, and more.

### What is etcd?
The etcd project (https://etcd.io/) is an incubating project on the Cloud Native and Computing Foundation Landscape (https://landscape.cncf.io/category=coordination-service-discovery&grouping=category). This is a distributed, reliable key-value store which mainly known because is heavily used by Kubernetes project (http://kubernetes.io/)

#### Some numbers
This is a widely used project and that can be inferred from its numbers:
- +33K start on github
- +500 contributors
- Important adopters like Kubernetes, ROOK, CoreDNS and OpenStack projects

### Current implementation
Today, the etcd project offers an official Java client to manage the etcd API. More details on https://github.com/etcd-io/jetcd

**Comment by graemerocher** [comment, marked as answer]
Sounds great, like other projects we encourage you to start in your own GitHub account and then when you are ready we can create the project and you can submit a PR it can be considered for inclusion. This is the same approach we took for https://github.com/micronaut-projects/micronaut-nats and https://github.com/micronaut-projects/micronaut-graphql which were contributed by external contributors initially

**Reply by alvarosanchez** [reply]
One thing to add to this, is to use https://github.com/micronaut-projects/micronaut-project-template as template for your repo.

**Reply by marcosflobo** [reply]
Alvaro, I guess I have to use that template and I'll work on my repo. Once I have something working, I'll have to ask about next steps

**Reply by marcosflobo** [reply]
Here it is https://github.com/marcosflobo/micronaut-etcd
